Feature/FIT-10-Details-Food

PeshoOReda opened the pull request:
This commit refactors the meal details functionality by moving related files and updating imports from the `meal` domain to the `meals` domain.

Specific changes include:
- Renamed `lib/domain/meal/entity/meal_details_entity.dart` to `lib/domain/meals/entity/meal_details_entity.dart`.
- Renamed `lib/domain/meal/repo/meal_details_repo.dart` to `lib/domain/meals/repo/meal_details_repo.dart`.
- Renamed `lib/domain/meal/use_case/meal_details_use_case.dart` to `lib/domain/meals/use_case/meal_details_use_case.dart`.
- Updated import paths in various files to reflect the new location of meal details related code.
- Adjusted the initial route in `inital_route_function.dart` from `AppRoutes.mealDetailsRoute` to `AppRoutes.loginRoute`.
- Updated `AppRoutes.mealDetailsRoute` to `AppRoutes.mealsDetailsRoute`.
- Ran `build_runner` to regenerate affected generated files.

<!--📌 Pull Request Template -->

<!--📝 Description -->

This template ensures that all pull requests adhere to the project's contribution standards. It enforces proper naming conventions, code quality, CI/CD validation, and testing requirements. Following this checklist helps maintain a clean and efficient codebase while ensuring compatibility with JIRA issue tracking.

## 📌 Pull Request Checklist

### ✅ General
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] The PR title follows the required format: `type/TA-<issue-number>-TaskName`.
- [x] The branch name follows the required format: `feature/meaningful-name`, `fix/issue-description`, etc.

### 📂 Code Quality
- [x] All filenames are in `snake_case`.
- [x] All functions and variable names follow Dart naming conventions.
- [x] All new code is formatted with `dart format .`

### 🛠 CI/CD Validation
- [x] Linting passes with `dart analyze`.
- [x] The branch name is validated against the allowed patterns.
- [x] The PR title is validated against the required format.

### 🚀 Testing
- [x] Unit tests are included for all new features and bug fixes.
- [x] `flutter test --coverage` has been run and the results reviewed.
- [x] The project builds successfully without errors.

### 📝 Additional Information
<!-- Provide any extra context or details about the changes in this PR. -->

### 🔗 Related JIRA Ticket(s):
- [x] JIRA Issue: `TA-<issue-number>`

### 📸 Screenshots (if applicable):
<!-- Add screenshots to show the changes visually. -->

PeshoOReda committed:
Refactor: Remove GitHub Actions workflows for branch name validation, Dart linting, Flutter unit tests, PR title validation, and SonarCloud analysis

feat: Add meal details feature with API integration
- Implemented MealsRetrofitClient for fetching meal details
- Created data source and repository for meal details
- Added use case for retrieving meal details
- Developed MealDetailsScreen UI with ingredients and nutrient information
- Introduced BluredContainer for enhanced UI design
- Updated routing to include meal details screen
- Added necessary models and DTOs for meal details

PeshoOReda committed:
feat: Update dependencies and add localization support

- Updated `flutter_lints` to version 6.0.0.
- Added `video_player` dependency for video playback functionality.
- Introduced `http_mock_adapter` and `mocktail` for improved testing capabilities.
- Generated localization keys for error messages and UI strings using `easy_localization`.
- Implemented `IngredientBody` and `IngredientItem` widgets for displaying meal ingredients.
- Created `MealHeaderVideo` widget to handle video playback with fallback image.
- Developed `NutrientBody` and `NutrientInfo` widgets for displaying nutritional information.
- Added unit tests for `MealsRetrofitClient`, `MealsRemoteDataSourceImpl`, and related use cases.
- Established `MealDetailsCubit` for state management in meal details screen.

PeshoOReda committed:
fix: Remove unused code in meals_retrofit_client test

PeshoOReda committed:
fix: Clean up code formatting and improve readability in various files

PeshoOReda committed:
Refactor meal details feature to use new details_food module

- Updated imports in meal_details_screen.dart to point to details_food module.
- Refactored ingredient_body.dart, ingredient_item.dart, meal_header_video.dart, nutrient_body.dart, and nutrient_info.dart to use the new details_food module.
- Modified meal_details_state.dart and meals_details_cubit.dart to include mealId in the state management.
- Created new meal_details_screen.dart, ingredient_body.dart, ingredient_item.dart, meal_header_video.dart, nutrient_body.dart, and nutrient_info.dart files in the details_food module.
- Updated tests to reflect changes in state management and imports.

PeshoOReda committed:
feat: Implement meal details feature with Retrofit client and data source

PeshoOReda committed:
refactor(meal_details): move meal details functionality to meals feature

This commit refactors the meal details functionality by moving related files and updating imports from the `meal` domain to the `meals` domain.

Specific changes include:
- Renamed `lib/domain/meal/entity/meal_details_entity.dart` to `lib/domain/meals/entity/meal_details_entity.dart`.
- Renamed `lib/domain/meal/repo/meal_details_repo.dart` to `lib/domain/meals/repo/meal_details_repo.dart`.
- Renamed `lib/domain/meal/use_case/meal_details_use_case.dart` to `lib/domain/meals/use_case/meal_details_use_case.dart`.
- Updated import paths in various files to reflect the new location of meal details related code.
- Adjusted the initial route in `inital_route_function.dart` from `AppRoutes.mealDetailsRoute` to `AppRoutes.loginRoute`.
- Updated `AppRoutes.mealDetailsRoute` to `AppRoutes.mealsDetailsRoute`.
- Ran `build_runner` to regenerate affected generated files.

Ahmed-Abdelghany-11 commented on lib/core/utils/routes/app_routes.dart:
```diff
@@ -10,6 +12,10 @@ class AppRoutes {
-  static Map<String, Widget Function(BuildContext)> routes = {};
+  static Map<String, Widget Function(BuildContext)> routes = {
+    // ignore: prefer_const_constructors
+    mealDetailsRoute: (context) => MealDetailsScreen(mealId: '52772'),
```
putting a fixed id is wrong

Ahmed-Abdelghany-11 commented on lib/data/meals/model/meal_details_response_dto.dart:
```diff
@@ -0,0 +1,221 @@
+}
+
+@JsonSerializable()
+class MealDetailsDto {
```
don't make two model classes in the same file

Ahmed-Abdelghany-11 commented on lib/domain/meals/entity/meal_details_entity.dart:
```diff
@@ -0,0 +1,44 @@
+      strYoutube != null && strYoutube!.contains('http');
+}
+
+extension MealDetailsEntityNutrients on MealDetailsEntity {
```
make file for extensions in core

Ahmed-Abdelghany-11 commented on lib/domain/meals/entity/meal_details_entity.dart:
```diff
@@ -0,0 +1,44 @@
+  });
+}
+
+class Ingredient {
```
separate this class in a new file

Ahmed-Abdelghany-11 commented on lib/features/details_food/presentation/view_model/cubit/meals_details_cubit.dart:
```diff
@@ -0,0 +1,39 @@
+        MealDetailsState(mealDetailsStatus: BaseInitialState(), mealId: mealId),
+      );
+
+  Future<void> getMealDetails(String id) async {
```
make this function private and make use GetMealDetailsAction to access it

Ahmed-Abdelghany-11 commented on lib/features/details_food/presentation/view/meal_details_screen.dart:
```diff
@@ -0,0 +1,83 @@
+    final theme = Theme.of(context);
+    return BlocProvider(
+      create: (context) =>
+          getIt<MealDetailsCubit>(param1: mealId)..getMealDetails(mealId),
```
use GetMealDetailsAction to access the function not like that

Nobel1893 commented on test/data/meal/api/meals_retrofit_client.dart:
remove this file

Nobel1893 reviewed (changes requested):
why did you remove the the github actions files. 
Retrofit client doesn't need unit testing

Nobel1893 commented:
![image](https://github.com/user-attachments/assets/d7c07634-711c-48b7-b2a3-6815da2b5c7a)

the status of those actions doesn't reported because u removed the actions files

PeshoOReda committed:
refactor(meal_details): move meal details functionality to meals feature

This commit refactors the meal details functionality by moving related files and updating imports from the `meal` domain to the `meals` domain.

Specific changes include:
- Renamed `lib/domain/meal/entity/meal_details_entity.dart` to `lib/domain/meals/entity/meal_details_entity.dart`.
- Renamed `lib/domain/meal/repo/meal_details_repo.dart` to `lib/domain/meals/repo/meal_details_repo.dart`.
- Renamed `lib/domain/meal/use_case/meal_details_use_case.dart` to `lib/domain/meals/use_case/meal_details_use_case.dart`.
- Updated import paths in various files to reflect the new location of meal details related code.
- Adjusted the initial route in `initial_route_function.dart` from `AppRoutes.mealDetailsRoute` to `AppRoutes.loginRoute`.
- Updated `AppRoutes.mealDetailsRoute` to `AppRoutes.mealsDetailsRoute`.
- Ran `build_runner` to regenerate affected generated files.

PeshoOReda committed:
refactor(meal_details): move meal details functionality to meals feature

This commit refactors the meal details functionality by moving related files and updating imports from the `meal` domain to the `meals` domain.

Specific changes include:
- Renamed `lib/domain/meal/entity/meal_details_entity.dart` to `lib/domain/meals/entity/meal_details_entity.dart`.
- Renamed `lib/domain/meal/repo/meal_details_repo.dart` to `lib/domain/meals/repo/meal_details_repo.dart`.
- Renamed `lib/domain/meal/use_case/meal_details_use_case.dart` to `lib/domain/meals/use_case/meal_details_use_case.dart`.
- Updated import paths in various files to reflect the new location of meal details related code.
- Adjusted the initial route in `initial_route_function.dart` from `AppRoutes.mealDetailsRoute` to `AppRoutes.loginRoute`.
- Updated `AppRoutes.mealDetailsRoute` to `AppRoutes.mealsDetailsRoute`.
- Ran `build_runner` to regenerate affected generated files.

EslamMagdy12 committed:
feat(meal_details): enhance meal details screen with argument handling and state management

EslamMagdy12 committed:
Merge remote-tracking branch 'origin/development' into feature/fit-10-details-food

# Conflicts:
#	assets/translations/ar.json
#	assets/translations/en.json
#	lib/core/assets/app_icons.dart
#	lib/core/utils/constants.dart
#	lib/core/utils/datasource_excution/api_constants.dart
#	lib/core/utils/di/di.config.dart
#	lib/core/utils/l10n/locale_keys.g.dart
#	lib/core/utils/routes/app_routes.dart
#	lib/data/meals/api/meals_retrofit_client.dart
#	lib/data/meals/api/meals_retrofit_client.g.dart
#	pubspec.lock
#	pubspec.yaml

EslamMagdy12 committed:
feat(meal-details): add meal details screen and related components

EslamMagdy12 committed:
fix(auth-repo): correct expected result type in registration test

EslamMagdy12 committed:
fix(ingredient-body): remove unnecessary check for empty ingredients

EslamMagdy12 committed:
feat(meal-navigation): add back button to meal image and refactor meal details layout
